File: Typscript/Step-1 Basic/Maths/checkPalindrome.ts
// Time Complexity: O(Log10(N))

function isPalindrome(n:number):boolean {
    let dup = n;
    let reverse = 0;
    while(n > 0) {
        let lastDigit = n%10;
        reverse = reverse * 10 + lastDigit
        n = Math.floor(n/10);
    }
    return dup == reverse;
}

console.log(isPalindrome(123))
console.log(isPalindrome(121))

File: Go/Step1/Maths/checkPalindrome.go
package main

// Time Complexity: O(Log10N)

func isPalindrome(num int) bool {
	dup := num
	rev := 0
	for num > 0 {
		lastDigit := num % 10
		rev = rev*10 + lastDigit
		num = num / 10
	}

	return dup == rev
}
